README.md

```diff
@@ -3,38 +3,45 @@ This repository contains the experimental work for the Bachelor Thesis by Vincen
 ## Prerequisites
-As a prerequisite you should have an installation of Anaconda or MiniConda on you device, check this link for more details https://docs.conda.io/projects/conda/en/latest/
+It is recommanded to have an installation of a virtual environment tool on your device, such as Conda, Pipenv, VirtualEnv.
 ## Installation
-Use the terminal or an Anaconda Prompt for the following steps:
-1. Create the environment from the environment.yml file:
-```c
-conda env create -f environment.yml
-```
-2. Activate the new environment: 
-```c
-conda activate myenv
+Use the terminal or a virtual environment manager Prompt for the following steps:
+1. Install the dependencies using the `requirements.txt" file:
+- Windows: 
+```bash
+py -m pip install -r requirements.txt
 ```
-3. Verify that the new environment was installed correctly:
-```c
-conda env list
+- Unix/MacOS:
+```bash
+python -m pip install -r requirements.txt
 ```
-At this point you should have succesfully created the virtual environment with the right dependencies<br>
-Now it is time to create the kernel on which your notebook will run, open the terminal and type:
-```c
+2. At this point you should have succesfully created the virtual environment with the right dependencies<br>
+Now it is time to create the kernel on which your notebook will run, for example, in a Conda environment, open the terminal and type:
+```bash
 python -m ipykernel install --user --name myenv
 ```
-Now just select this kernel you created when running the NotebookAnalysisResults notebook.
-## Run the web API
-Use the terminal or an Anaconda Prompt to perform the following steps:
-1. Move to the folder containing the `main.py` module
-2. Start the API server with the following command:
-```bash
-uvicorn main:app --reload
+for more help on this matter, check this link: https://ipython.readthedocs.io/en/stable/install/kernel_install.html<br>
+Now just select this kernel you created when running the `Using pynblint.ipynb` notebook.<br><br>
+`pynblint` is a distributable package, it is installable thanks to its `setup.py` file:<br>
+1. cd into the root directory where setup.py is located 
+2. Enter: 
+```bash 
+python setup.py install
+```
+## Generate the documentation
+In order to make a firt build of the docs, let's say that you want to build a HTML documentation, type:
+```bash 
+sphinx-build -b html <source_directory> <builddir>
 ```
-The reload command makes the server restart after code changes. Only do this for development.<br>
-3. If the server starts correctly, the address and port on which it is running will be printed on your terminal.
+and then:
+```bash 
+make html
+```
+This will build HTML docs in the build directory you chose. Execute make without an argument to see which targets are available.<br>
```

Comment:
```suggestion
This will build HTML docs in the build directory you chose. Execute `make` without an argument to see which targets are available.<br>
```